core/src/main/scala/cromwell/core/path/EvenBetterPathMethods.scala

```diff
@@ -101,24 +101,34 @@ trait EvenBetterPathMethods {
     write(content)(openOptions, Codec.UTF8)
   }
 
-  /*
-   * The input stream will be closed when this method returns, which means the f function
-   * cannot leak an open stream.
-   */
-  def withReader[A](f: BufferedReader => A)(implicit ec: ExecutionContext): A = {
+  private def fileIoErrorPf[A]: PartialFunction[Throwable, Try[A]] = {
+    case ex: Throwable => Failure(new IOException(s"Could not read from ${this.pathAsString}: ${ex.getMessage}", ex))
+  }
```

Comment:
Does this really have to be partial? Would it work if this was just a regular function from `Throwable` to `Try[A]`?